src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java

```diff
@@ -475,39 +472,26 @@ private NotificationMessage doSync(FormplayerSyncScreen screen) throws Exception
      * <p>
      * Will do nothing if this wasn't a query screen.
      */
-    private NotificationMessage doQuery(FormplayerQueryScreen screen,
-                                        Hashtable<String, String> queryDictionary,
-                                        boolean skipDefaultPromptValues) {
+    private void doQuery(FormplayerQueryScreen screen,
+                         Hashtable<String, String> queryDictionary,
+                         boolean skipDefaultPromptValues) throws CommCareSessionException {
         log.info("Formplayer doing query with dictionary " + queryDictionary);
         if (queryDictionary != null) {
             screen.answerPrompts(queryDictionary);
         }
 
-        String error = null;
         ExternalDataInstance searchDataInstance;
         try {
             searchDataInstance = searchAndSetResult(
                     screen,
                     screen.getUri(skipDefaultPromptValues));
             if (searchDataInstance == null) {
-                error = "No result from query";
+                throw new CommCareSessionException("No result from query");
```

Comment:
Switching to an exception doesn't change the user-facing behavior, does it?

Reply:
no, this would have resulted in an exception before anyway: https://github.com/dimagi/formplayer/blob/49a245a82ef33f61a05394a965c60438ddc1f7ac/src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java#L349-L357

The exception handling is done here: https://github.com/dimagi/formplayer/blob/e638a9e628b470ce9d0df2571f1b748a815d4a5c/src/main/java/org/commcare/formplayer/application/AbstractBaseController.java#L110-L125